src/foam/dao/index/Index.js

```diff
@@ -15,52 +15,114 @@
  * limitations under the License.
  */
 
-/* Indexed Memory-based DAO. */
+/**
+  The Index interface for an ordering, fast lookup, single value,
+  index multiplexer, or any other MDAO select() assistance class.
 
-/*
- * TODO:
- *  update(oldValue, newValue)
- *  reuse plans
- *  add ability for indices to pre-populate data
- */
+  Each Index subclass also defines an IndexNode class. Index defines
```

Comment:
Every index has nodes, since any index may be the tail of a Tree (or other type that splits into many sub-nodes). Indexes always worked this way, but now the shared definition object exists as Index, and the existing data-searching/manipulating methods moved to IndexNode.